cwizard.py

```diff
@@ -0,0 +1,124 @@
+
+    # Question 1
+    def choose_source(self):
+        """Asks the user for the data source"""
+        print("\nDo you want to send a dictionary or a textfile?")
+        response = input("d/t: ")
+        match response:
+            case "d":
+                print("You selected dictionary")
+                self.source = Source.Dictionary
+            case "t":
+                print("You selected textfile")
+                self.source = Source.TextFile
+            case _:
+                print("Invalid source")
+                self.choose_source()
+
+    def choose_format(self):
+        """Asks the user for the pickling format"""
+        print("\nChoose a pickling format: Binary, JSON, or XML")
+        response = input("b/j/x: ")
+        match response:
+            case "b":
+                print("You selected binary")
+                self.format = Format.BINARY
+            case "j":
+                print("You selected json")
+                self.format = Format.JSON
+            case "x":
+                print("You selected XML")
+                self.format = Format.XML
+            case _:
+                print("Invalid pickling format")
+                self.choose_format()
+
+    def choose_textfile(self):
+        """Asks the user to select the filepath to the textfile"""
+        print("\nPlease input the filepath to the textfile")
+        response = input("filepath: ")
+        if os.path.exists(response):
+            self.filepath = response
+        else:
+            print("Invalid filepath")
+            self.choose_textfile()
+
+    def choose_security(self):
+        """Asks the user to select the security option"""
+        print("\nDo you want to encrypt the file?")
+        response = input("y/n: ")
+        match response:
+            case "y":
+                print("You chose to encrypt the file")
+                self.security = SecurityLevel.Encrypted
+            case "n":
+                print("You chose not to encrypt the file")
+                self.security = SecurityLevel.UnEncrypted
+            case _:
+                print("Invalid security level")
+                self.choose_security()
+
+    def build_dictionary(self):
+        """Builds a dictionary item by item"""
+        print("\nNow we will build the dictionary")
+        item = 1
+        flag = True
+        while flag:
+            key = input(f"\nItem {item} key: ")
+            value = input(f"Item {item} value: ")
+            self.dictionary.update({key: value})
+            response = input("type 'x' if that was your last item: ")
+            if response == "x":
+                flag = False
+            else:
+                item += 1
+
+    def ask_all(self):
+        """Runs the Wizard"""
+        print("\n*** Welcome to the client Wizard ***")
+        self.choose_source()
+        # Dictionary
+        if self.source == Source.Dictionary:
+            self.choose_format()
+            self.build_dictionary()

```

Comment:
we need to add the security for the dictionary also

Reply:
The task states

Create a dictionary, populate it, serialize it and send it to a server.
Create a text file and send it to a server.
With the dictionary, the user should be able to set the pickling format to one of the following: binary, JSON and XML. Also, the user will need to have the option to encrypt the text in a text file.

So 
1) dictionary needs to be capable of being serialised 3 ways,
2) text file needs to be capable of being encrypted

No option is required of being serialised and encrypted.

Did you read it as requiring both?

Reply:
Yesterday I thought we agreed on doing it for both. Even if the teacher doesn't state it directly, it would benefit the User to do it. It is no real extra effort if we implement it with the suggested protocol:

![image](https://github.com/mr-mdd/CSCK541---EOM---Group-A/assets/102789861/ec05bb22-2198-4336-9f68-e8987d2d0e52)


And follow the flow chart that I thought was agreed upon:
![image](https://github.com/mr-mdd/CSCK541---EOM---Group-A/assets/102789861/26f20fd8-2fa7-4a09-b155-111cbcb558f5)

But if you wish to continue with that understanding, I will just update the diagrams